build_projects/shared-build-targets-utils/Utils/PublishMutationUtilties.cs

```diff
@@ -22,7 +22,7 @@ public class PublishMutationUtilties
             foreach (var binaryName in binariesToBeDeleted)
             {
                 File.Delete(Path.Combine(path, $"{binaryName}{Constants.ExeSuffix}"));
-                File.Delete(Path.Combine(path, $"{binaryName}.dll"));
+                File.Delete(Path.Combine(path, $"{Constants.DynamicLibPrefix}{binaryName}{Constants.DynamicLibSuffix}"));
```

Comment:
I dont think this is used for managed - here is the only usage that works on native binaries:

```
 PublishMutationUtilties.CleanPublishOutput(
                sharedFrameworkNameAndVersionRoot,
                "framework",
                new List<string> {"apphost", "hostfxr", "dotnet" },
                deleteRuntimeConfigJson: true,
                deleteDepsJson: false);
```